provisioning_templates/finish/preseed_default_finish.erb

```diff
@@ -15,10 +15,12 @@ oses:
 %>
 
 <% subnet = @host.subnet -%>
-<% if subnet.respond_to?(:dhcp_boot_mode?) -%>
-<% dhcp = subnet.dhcp_boot_mode? && !@static -%>
+<% if @static -%>
+  <%- dhcp = false -%>
+<% elsif subnet.nil? -%>
+  <%- dhcp = true -%>
 <% else -%>
-<% dhcp = !@static -%>
+  <%- dhcp = subnet.respond_to?(:dhcp_boot_mode?) && subnet.dhcp_boot_mode? -%>
```

Comment:
Why would subnet not respond to `dhcp_boot_mode`? It's either subnet class or `nil`, you can use the `subnet&.dhcp_boot_mode?` syntax to cover nil case but that's pretty much all I can see. Did you encounter different object than subnet there?

Reply:
To be honest, this is my first foray into Ruby and Foreman. The `subnet.respond_to?(:dhcp_boot_mode?)` check was there previously so I've just maintained it. If, as you say, it's always a subnet and it's already be checked for `nil` then I presume this check can be removed?

Reply:
```suggestion
  <%- dhcp = subnet.dhcp_boot_mode? -%>
```

Reply:
Committed all `subnet.respond_to?(:dhcp_boot_mode?)` checks removed.